src/Dependencies/Collections/SegmentedList`1.cs

```diff
@@ -855,6 +873,36 @@ public int LastIndexOf(T item, int index, int count)
             return SegmentedArray.LastIndexOf(_items, item, index, count);
         }
 
+        public int LastIndexOf(T item, int index, int count, IEqualityComparer<T>? comparer)
+        {
+            if ((Count != 0) && (index < 0))
```

Comment:
nit: the parentheses look redundant.

```suggestion
            if (Count != 0 && index < 0)
```

Reply:
➡️ Going to leave these per https://github.com/dotnet/roslyn/pull/54424#discussion_r659511408